openapi/src/parts/controlled_aws_bucket.yaml

```diff
@@ -0,0 +1,146 @@
+paths:
+  /api/workspaces/v1/{workspaceId}/resources/controlled/aws/buckets:
+    parameters:
+      - $ref: '#/components/parameters/WorkspaceId'
+    post:
+      summary: Create a new controlled AWS bucket
+      operationId: createAwsBucket
+      tags: [ControlledAwsResource]
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateControlledAwsBucketRequestBody'
+      responses:
+        '200':
+          $ref: '#/components/responses/CreatedControlledAwsBucketResponse'
+        '403':
+          $ref: '#/components/responses/PermissionDenied'
+        '500':
+          $ref: '#/components/responses/ServerError'
+
+  /api/workspaces/v1/{workspaceId}/resources/controlled/aws/buckets/{resourceId}:
+    parameters:
+      - $ref: '#/components/parameters/WorkspaceId'
+      - $ref: '#/components/parameters/ResourceId'
+    get:
+      summary: Get a controlled AWS bucket resource
+      operationId: getAwsBucket
+      tags: [ControlledAwsResource]
+      responses:
+        '200':
+          $ref: '#/components/responses/GetControlledAwsBucketResponse'
+        '403':
+          $ref: '#/components/responses/PermissionDenied'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '500':
+          $ref: '#/components/responses/ServerError'
+
+  /api/workspaces/v1/{workspaceId}/resources/controlled/aws/buckets/{resourceId}/consoleLink:
```

Comment:
I'd like to consider having a general `/credential' endpoint with a parameter or body element that says what sort of credential you want. My concern is that it will not end with just `credential` and `consoleLink`, but there will be credentials for this tool and that tool.